src/GraphQL.Tests/Types/InterfaceTests.cs

```diff
@@ -0,0 +1,110 @@
+using GraphQL.Types;
+
+namespace GraphQL.Tests.Types;
+
+public class InterfaceTests : QueryTestBase<InterfaceTests.MySchema>
+{
+    [Fact]
+    public void InterfaceWithNoObject()
+    {
+        AssertQuerySuccess(
+            """
+            {
+                noObjectTest {
+                    id                    # as there is no object type defined for this
```

Comment:
```suggestion
                    __typename
                    id                    # as there is no object type defined for this
```

try this

Reply:
Right now it crashes with a null reference exception.